### PR TITLE
SONARJS-432 Improve rule (S2508) The names of model properties should not contains spaces

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/SpaceInModelPropertyNameCheck.java [hunk 1]
```diff
@@ -46,34 +49,62 @@
...
-      if (key.is(Kind.STRING_LITERAL) && StringUtils.indexOf(((LiteralTree) key).value(), ' ') >= 0) {
-        getContext().addIssue(this, key, "Rename this property to remove the spaces.");
+      if (defaultsProp != null && defaultsProp.value().is(Kind.OBJECT_LITERAL)) {
+        checkForSpaceInPropertyNames((ObjectLiteralTree) defaultsProp.value());
       }
     }
   }
 
+  private boolean isBackboneSetMethod(DotMemberExpressionTree dotExpr) {
+    return CheckUtils.asString(dotExpr.property()).equals("set") && dotExpr.object().types().contains(Type.Kind.BACKBONE_MODEL_OBJECT);
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/SpaceInModelPropertyNameCheck.java]: MAJOR Move the "set" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 major

**Comment by lindamartin** [comment]
@vilchik-elena looks like this PR includes all the commits from https://github.com/SonarCommunity/sonar-javascript/pull/22

**Comment by lindamartin** [comment]
@vilchik-elena  looks good to me!

**Pull request merged**